### ISSUE TITLE
Roadmap question: Compatibility with registered users?

### ISSUE THREAD

**Issue by Rello** [SergeyMosin/Appointments#56]
Hello,

as the impact of my question is rather complex, I would like to know if you thought about it already for the future:

**As Is**
- The app is aimed at external requestors. If an internal user uses it, it could be more integrated

**Proposal**
- no need to enter contact data
- automatic addition into calendar
- sharing appointment "public page" with user/usergroup

**use case:**
- Teacher & Classes

what is your opinion on this feature?

**Comment by SergeyMosin** [comment]
I'll have to look in to this, but this was really made for "external" purposes.

**Comment by Rello** [comment]
I am happy to close the issue if you don´t want to have your issue-log spammed...

**Comment by SergeyMosin** [comment]
No, leave it open. Someone might come along and contribute some code if it gains traction.

**Comment by SergeyMosin** [comment]
"Private" mode is available in v1.12.0 more info https://github.com/SergeyMosin/Appointments/issues/298#issuecomment-1047218793

**Comment by JocelynDelalande** [comment]
@SergeyMosin I was thinking : maybe the easy (and helpfull) part of that feature request would be the following items :

- if logged-in user and private mode : do not show the name and email fields
- fill those « fields » from NC logged-in user data

**Comment by CSDUMMI** [comment]
I'd be willing to look into implementing a feature to fill-in the contact info from the current user into the email and telephone field (if present). 

I believe this ought to be a pre-filled field that the user can subsequently review and chnage if they wish.

**Comment by JocelynDelalande** [comment]
I think « locked » filling also has a use, as it allows registered booking. Anyway, wether the email or phone are « forced », it could be a good idea to store the user ID / user handle on the booked the event, maybe in a ical event property ?
